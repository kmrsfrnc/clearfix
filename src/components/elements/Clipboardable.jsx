import React from 'react';
import PropTypes from 'prop-types';
import Clipboard from 'react-clipboard.js';
import Tooltip from '@material-ui/core/Tooltip';

class Clipboardable extends React.Component {
  static propTypes = {
    value: PropTypes.string,
    children: PropTypes.node,
  };

  static defaultProps = {
    value: '',
    children: '',
  };

  state = {
    isSuccess: false,
  };

  toggleIsSuccess = (callback) => {
    this.setState(({ isSuccess }) => ({
      isSuccess: !isSuccess,
    }), callback);
  };

  onSuccess = () => {
    this.toggleIsSuccess(() => {
      setTimeout(this.toggleIsSuccess, 1500);
    });
  };

  render() {
    const { children, value, ...otherProps } = this.props;
    const { isSuccess } = this.state;

    return (
      <Tooltip placement="right" open={isSuccess} title="Copied" {...otherProps}>
        <Clipboard data-clipboard-text={value} onSuccess={this.onSuccess}>
          {children}
        </Clipboard>
      </Tooltip>
    );
  }
}

export default Clipboardable;
