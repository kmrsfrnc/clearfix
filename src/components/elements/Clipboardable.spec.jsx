import React from 'react';
import { shallow } from 'enzyme';

import Clipboardable from './Clipboardable';

jest.useFakeTimers();

describe('Clipboardable', () => {
  it('matches snapshot', () => {
    expect(shallow(<Clipboardable value="test">test</Clipboardable>)).toMatchSnapshot();
  });

  it('passes other props', () => {
    const otherProps = { foo: 'bar' };
    const wrapper = shallow(<Clipboardable value="test" {...otherProps}>test</Clipboardable>);

    expect(wrapper.props().foo).toEqual('bar');
  });

  it('shows a tooltip for some time on success', () => {
    const wrapper = shallow(<Clipboardable value="test">test</Clipboardable>);

    const { onSuccess } = wrapper.childAt(0).props();
    onSuccess();

    expect(wrapper.props().open).toBeTruthy();
    jest.runOnlyPendingTimers();
    expect(wrapper.props().open).toBeFalsy();
  });
});
