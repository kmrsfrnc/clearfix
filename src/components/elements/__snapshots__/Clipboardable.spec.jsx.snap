// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clipboardable matches snapshot 1`] = `
<WithStyles(Tooltip)
  open={false}
  placement="right"
  title="Copied"
>
  <ClipboardButton
    data-clipboard-text="test"
    onClick={[Function]}
    onSuccess={[Function]}
  >
    test
  </ClipboardButton>
</WithStyles(Tooltip)>
`;
